plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -264,7 +268,7 @@ public TaskStatus process()
                 // 2. Timeout after waiting for the configured time
                 long timeLeft = dynamicFilteringWaitTimeoutMillis - stopwatch.elapsed(MILLISECONDS);
                 if (timeLeft > 0 && dynamicFilter.isAwaitable()) {
-                    future = asVoid(toListenableFuture(dynamicFilter.isBlocked().orTimeout(timeLeft, MILLISECONDS)));
+                    future = asVoid(Futures.withTimeout(toListenableFuture(dynamicFilter.isBlocked()), timeLeft, MILLISECONDS, timeoutExecutor));
```

Comment:
alternatively you could use
```
.thenApply(Function.identity())
```
(with comment) then you wouldn't have to deal with executor

Reply:
I didn't understand this comment, thenApply on which future?

Reply:
```
dynamicFilter.isBlocked().thenApply(Function.identity()).orTimeout
```

Reply:
Got you. Fixed. Much simpler indeed. Thanks.